specutils/io/default_loaders/sdss.py

```diff
@@ -113,38 +159,96 @@ def spSpec_loader(file_obj, **kwargs):
     Returns
     -------
     data: Spectrum1D
-        The spectrum that is represented by the data in this table.
+        The spectrum that is represented by the wavelength solution from the
+        header WCS and data array of the primary HDU.
     """
     if isinstance(file_obj, fits.hdu.hdulist.HDUList):
         hdulist = file_obj
     else:
         hdulist = fits.open(file_obj, **kwargs)
 
     header = hdulist[0].header
-    name = header.get('NAME')
+    # name = header.get('NAME')
     meta = {'header': header}
-    wcs = WCS(hdulist[0].header)
+    wcs = WCS(header).dropaxis(1)
+
+    bunit = header.get('BUNIT', '1e-17 erg / (Angstrom cm2 s)')
+    # fix mutilated flux unit
+    bunit = bunit.replace('/cm/s/Ang', '/ (Angstrom cm2 s)')
+    if 'Ang' in bunit and 'strom' not in bunit:
+        bunit = bunit.replace('Ang', 'Angstrom')
+    flux_unit = Unit(bunit)
+    flux = hdulist[0].data[0, :] * flux_unit
+
+    uncertainty = StdDevUncertainty(hdulist[0].data[2, :] * flux_unit)
+
+    # dispersion along NAXIS1 from the WCS
+    dispersion = wcs.pixel_to_world(np.arange(flux.shape[0]))
+    # convert out of logspace (default for spSpec/spPlate spectra)?
+    if header.get('DC-Flag', 1) == 1:
+        dispersion = 10**dispersion
+    dispersion_unit = Unit('Angstrom')
 
-    data = hdulist[0].data[0, :]
-    unit = Unit('1e-17 erg / (Angstrom cm2 s)')
+    mask = hdulist[0].data[3, :] != 0
 
-    uncertainty = StdDevUncertainty(hdulist[0].data[2, :] * unit)
+    if not isinstance(file_obj, fits.hdu.hdulist.HDUList):
+        hdulist.close()
 
-    # dispersion from the WCS but convert out of logspace
-    # dispersion = 10**wcs.all_pix2world(np.arange(data.shape[0]), 0)[0]
-    dispersion = 10**wcs.all_pix2world(np.vstack((np.arange(data.shape[0]),
-                                                  np.zeros((data.shape[0],)))).T,
-                                       0)[:, 0]
-    # dispersion = 10**hdulist[1].data['loglam']
+    return Spectrum1D(flux=flux, spectral_axis=dispersion * dispersion_unit,
+                      uncertainty=uncertainty, meta=meta, mask=mask)
+
+
+@data_loader(label="SDSS spPlate", identifier=spPlate_identify, extensions=['fits'])
+def spPlate_loader(file_obj, limit=None, **kwargs):
```

Comment:
OK; I had thought if leaving that in it would actually make more sense to provide both `start` and `stop`, or a pair of `limits=[]`, so the user could extract a range of spectra. Only point would be some memory savings, I guess, but at 60 MB for the entire collection of spectra this should not really matter.